sdk/core/core/inc/az_static_assert.h

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_STATIC_ASSERT_H
+#define AZ_STATIC_ASSERT_H
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define AZ_STATIC_ASSERT(CONDITION) typedef int _az_static_assert[(CONDITION) ? 1 : -1];
```

Comment:
I believe that names beginning with a leading underscore are reserved for the standard library, so while I doubt there is actual worry for collisions we should try to avoid this where possible.

Reply:
@barcharcraz can we use `_` for private/internal things?